### PR TITLE
Reactor 2.0. Introduce email effect

### PR DESCRIPTION
**Pull request by dnegma**
### Context
This is adding the functionality to the email handler to actually send the email when receiving the `:notify/email` event. Note that producing the real email content is not part of this PR, for now just sending some random string to verify it works.

https://app.asana.com/0/736811264035867/769735018300330

### Changes
**New namespace `blueprints.reactor.deps`**
Copied over necessary mail deps from old reactor to be able to use mocks, and dev configs on services with side effects. Definitely open to suggestions on how to handle different environments here. I'm just merging for now.

**Added lib: `expound`**
Library for much better clojure.spec error messages to understand what's going wrong. [Github repo](https://github.com/bhb/expound)

**New effect: `:channel`**
I introduced the new effect channel, so we can handle channels (which is used by email and slack to send async messages). The effect will just make sure take the value from the channel and throw an exception if it found one.
```clojure
(defmethod effect!* :channel
  [_ _ _ ch]
  ;; Is valid content on channel? Otherwise throws exception.
  (tb-async/<!!? ch)
  ;; Return nil on success, no more effects caused.
  nil)
```

### CODE REVIEW

File: src/blueprints/reactor/deps.clj [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   :public-hostname    (config/root-domain config)})
+
+
+(defn deps [config & [{:keys [use-mocks?]}]]
+  (cond-> (base-deps config)
+
+          (false? use-mocks?)
+          (merge (dev-deps config))
+
+          #_#_(config/production? config)
```
Comment (allenjprice) [inline src/blueprints/reactor/deps.clj]: It seems like we don't need this - is this a placeholder for later?

File: src/blueprints/reactor.clj [hunk 2]
```diff
@@ -169,7 +170,9 @@
 
 
 (defstate reactor
-  :start (start! {:teller teller
-                  :config config
-                  :conn   conn})
+  :start (start! (merge (r-deps/deps config
+                                     {:use-mocks? true})
```
Comment (allenjprice) [inline src/blueprints/reactor.clj]: Under what conditions would `:use-mocks?` ever be false?

File: src/blueprints/reactor/deps.clj (hunk 1 above)
Comment (dnegma) [reply src/blueprints/reactor/deps.clj]: Yah, it's for production configs (in dev we override certain things, like sending all emails and posting all slack messages to one place)

**Comment by allenjprice** [comment]
Looks great, @dnegma! I have a couple of questions that I posted as comments, but I wouldn't consider these blockers to merging.

File: src/blueprints/reactor.clj (hunk 2 above)
Comment (dnegma) [reply src/blueprints/reactor.clj]: It's just a flag for dev when you want to test the side effects for some reason, such as email and slack. (I've done that to make sure things are being sent). I thought it seems unnecessary to have it enabled by default and have a bunch of stuff sent, if it's not part of what you're testing. Makes sense?

Comment (allenjprice) [reply src/blueprints/reactor.clj]: Wait, disregard this question because it is dumb and the answer to it is clearly "when we're using it in production."

Comment (dnegma) [reply src/blueprints/reactor.clj]: Lol, could be, but I don't think this should have to be touched for production to work properly. Good point, the production check should probably come before checking for this (when introduced as it's currently commented out)

**Pull request merged**